### ISSUE TITLE
RegEx used by GitHub actions to automatically label pull-requests requires tweaking

### ISSUE THREAD

**Issue by ghuntley** [unoplatform/uno#1788]
## Current behavior

https://github.com/unoplatform/uno/pull/1787

## Expected behavior

- Pull-requests to be automatically labeled as per `.github/labeler.yml` and https://github.com/unoplatform/uno/pull/1773

## How to reproduce it (as minimally and precisely as possible)

See https://github.com/unoplatform/uno/pull/1787 and https://github.com/actions/labeler

## Anything else we need to know?

Opening this as a devops task for hacktoberfest. Copy the contents of this repository under your personal account that has GitHub actions enabled upon it and iterate. Alternatively iterate locally using the GitHub actions docker image/developer environment. Once you have found the solution please send back a pull-request <3

**Comment by jotaro-sama** [comment]
> iOS, Android and WASM labels were not automatically applied.

Is this the issue to be addressed or is there more? I'll do what I can to find what is causing the issue.

**Comment by jotaro-sama** [comment]
> iOS, Android and WASM labels were not automatically applied.
Is this the only issue to be addressed, or is there something else as well?

Anyway, I guess I'll try figuring out what's causing the issue. I'll let you know and post a PR if I manage to fix this!

**Comment by ghuntley** [comment]
@jotaro-sama thanks so much for digging in <3.

> Is this the only issue to be addressed, or is there something else as well?

The RegEx for the other labels maybe incorrect as well, if you could check these whilst in there that would be 🌟

**Comment by jotaro-sama** [comment]
I solved the issue on my copy of your branch! 

You can see the PRs on my fork where all of the labelings are now working as intended, be them Android, iOS, WASM or whatever (except those on the hidden files and directories, whose filename starts with a dot, I think those are ignored by the labeler anyways). 

However, I had to:

1. Copy the contents of the `.github` folder from the main branch that were missing on `dev/ghuntley/automatic-release-notes`. I am by no means an expert of GitHub Actions, but I'm wondering: how did you even get the bot to add any labels at all on #1787 if there was neither `workflows/labeler.yml` nor `labeler.yml` inside the `.github` folder? AFAIK to start a workflow when a PR is created you have to have a `.yml` file describing the workflow inside the `workflows` folder, and GitHub looks for it in the head repository. If there's another way to start workflows I'm genuinely curious about it!
2. Solve a merge conflict with the master branch in order to get the workflow executed, so now it appears my copy of your branch is a lot of commits ahead of your repo. Sadly I had to make many useless commits just to do tests, since the whole thing was GitHub-related.

If I were to PR now into your branch directly, I think it would make a mess, so I was thinking about creating a new branch from `uno/dev/ghuntley/automatic-release-notes` and only edit the `labeler.yml` file (which was the only thing that actually needed changes). But I would still need to add the files from the `.github` folder that are missing in your repo. 

Are you fine with me doing that? Or would you prefer a PR directly to the master or some other branch?

**Comment by jotaro-sama** [comment]
I solved the issue on my copy of your branch! 

You can see the PRs on my fork where all of the labelings are now working as intended, be them Android, iOS, WASM or whatever (except those on the hidden files and directories, whose filename starts with a dot, I think those are ignored by the labeler anyways). 

However, I had to:

1. Copy the contents of the `.github` folder from the main branch that were missing on `dev/ghuntley/automatic-release-notes`. I am by no means an expert of GitHub Actions, but I'm wondering: how did you even get the bot to add any labels at all on #1787 if there was neither `workflows/labeler.yml` nor `labeler.yml` inside the `.github` folder? AFAIK to start a workflow when a PR is created you have to have a `.yml` file describing the workflow inside the `workflows` folder, and GitHub looks for it in the head repository. If there's another way to start workflows I'm genuinely curious about it!
2. Solve a merge conflict with the master branch in order to get the workflow executed, so now it appears my copy of your branch is a lot of commits ahead of your repo. Sadly I had to make many useless commits just to do tests, since the whole thing was GitHub-related.

Long story short, if I were to PR now into your branch directly, I think it would make a mess, so I was thinking about creating a new branch from `uno/dev/ghuntley/automatic-release-notes` and only edit the `labeler.yml` file (which was the only thing that actually needed changes). But I would still need to add the files from the `.github` folder that are missing in your repo. 

Are you fine with me doing that? Or would you prefer a PR directly to the master or some other branch?

**Comment by jotaro-sama** [comment]
I actually found out that there were some other incorrect RegExes, but now everything should work fine now, I've made all tests I could think of. I initially thought the `dev/ghuntley/automatic-release-notes` branch was the one where you wanted the modifications (I mistook it for the `dev/ghuntley/automatically-label-pull-requests` branch which is now deleted), so you can see all of my tests on my fork of that one if you're curious (e.g. https://github.com/jotaro-sama/uno/pull/7). I hope this will help you guys!
